Automattic/wp-calypso
# Launchpad: Add calypso_signup_complete to launchpad

### Proposed Changes

**NOTE: The methodology being used for this has changed since last round of reviews and approvals. Needs re-testing.**

Right now, the `calypso_signup_complete` tracking event is not firing for any stepper flows except newsletter. This PR makes the following changes:
 - Invoke recordSignupComplete() when someone lands on launchpad. This is the method martech/analytic would like us to use (see pau2Xa-4DX-p2)
 - Add a custom hook to encapsulate the logic for recordSignupComplete. We need to fetch several values for meta, and I didn't want to put all that extra code directly in launchpad/index.tsx.
 - To avoid duplicate events, store the siteSlug in local storage when calypso_signup_complete is fired for a site, and check that value before firing the event. So if you navigate away from launchpad and comes back, the event will not re-fire. 
 - Clear our localStorage value whenever someone clicks the 'primary action' button on Launchpad. For most flows, that's the Launch button, but it may vary. 

### Testing Instructions

**Test Time: Medium**
**Review Time: Medium**

**Setup**
1. Ensure you can see tracks events. You can do that by installing [Tracks Vigilante](https://github.com/Automattic/tracks-chrome-extension). Or open developer tools, go to network, and filter by `t.gif`.
2. Check out this branch and run yarn and yarn start if needed. 

**Test Link in Bio Flow**
1. Go to http://calypso.localhost:3000/setup/link-in-bio/intro and start a Link in Bio flow. 
2. Once you arrive on the Launchpad screen:
   - Check the tracks events and confirm that calypso_signup_complete has fired with the following meta: `flow`, `intent`, and `starting_point` all = the flow name; `theme` = the theme you chose; `is_blank_canvas` = false (unless you chose blank canvas); `is_new_user` = false (unless you logged in as a new user).
   - Check Dev Tools > Application > Local Storage and confirm you see the launchpad_siteSlug value stored there, and that it matches your new site slug. 
3. Click Add Links and return to Launchpad. Confirm that calypso_signup_complete does not re-fire.
4. Click the Launch button and confirm that the launchpad_siteSlut item is cleared from local storage. 

**Test Free Flow**
1. Go to http://calypso.localhost:3000/setup/free/intro and start a Free Flow. 
2. Once you arrive on the Launchpad screen:
   - Check the tracks events and confirm that calypso_signup_complete has fired with the following meta: `flow`, `intent`, and `starting_point` all = the flow name; `theme` = the theme you chose; `is_blank_canvas` = false (unless you chose blank canvas); `is_new_user` = false (unless you logged in as a new user).
   - Check Dev Tools > Application > Local Storage and confirm you see the launchpad_siteSlug value stored there, and that it matches your new site slug. 
3. Click Publish first post and return to Launchpad. Confirm that calypso_signup_complete does not re-fire.
3. Click Publish first post and return to Launchpad. Confirm that calypso_signup_complete does not re-fire.
4. Click the Launch button and confirm that the launchpad_siteSlut item is cleared from local storage. 

Related to https://github.com/Automattic/wp-calypso/issues/70883
